client/my-sites/plugins/plugins-list/index.jsx

```diff
@@ -0,0 +1,402 @@
+		if ( this.props.isWpCom !== true ) {
+			PluginsLog.removeListener( 'change', this.showDisconnectDialog );
+		}
+	},
+
+	isSelected( plugin ) {
+		return !! this.state.selectedPlugins[ plugin.slug ];
+	},
+
+	togglePlugin( plugin ) {
+		const { slug } = plugin;
+		const { selectedPlugins } = this.state;
+		const oldValue = selectedPlugins[ slug ];
+		this.setState( { selectedPlugins: Object.assign( {}, selectedPlugins, { [ slug ]: ! oldValue } ) } );
+		analytics.ga.recordEvent( 'Plugins', 'Clicked to ' + this.isSelected( plugin ) ? 'Deselect' : 'Select' + 'Single Plugin', 'Plugin Name', slug );
+	},
+
+	setBulkSelectionState( plugins, selectionState ) {
+		let slugsToBeUpdated = {};
+		plugins.forEach( plugin => slugsToBeUpdated[ plugin.slug] = this.hasNoSitesThatCanManage( plugin ) ? false : selectionState );
+
+		this.setState( { selectedPlugins: Object.assign( {}, this.state.selectedPlugins, slugsToBeUpdated ) } );
+	},
+
+	getPluginBySlug( slug ) {
+		for ( let i = 0, l = this.props.plugins.lenght; i < l; i++ ) {
+			if ( this.props.plugins[ i ].slug === slug ) {
+				return this.props.plugins[ i ];
+			}
+		}
+		return false;
+	},
+
+	filterSelection: {
+		active( plugin ) {
+			if ( this.isSelected( plugin ) ) {
+				return plugin.sites.some( site => site.plugin && site.plugin.active );
+			}
+			return false;
+		},
+		inactive( plugin ) {
+			if ( this.isSelected( plugin ) ) {
+				return plugin.sites.some( site => site.plugin && ! site.plugin.active );
+			}
+			return false;
+		},
+		updates( plugin ) {
+			if ( this.isSelected( plugin ) ) {
+				return plugin.sites.some( site => site.plugin && site.plugin.update && site.canUpdateFiles );
+			}
+			return false;
+		},
+		selected( plugin ) {
+			return this.isSelected( plugin );
+		}
+	},
+
+	getPluginUpdateCount() {
+		if ( this.props.isWpCom ) {
+			return 0;
+		}
+		return this.props.pluginUpdateCount;
+	},
+
+	hasNoSitesThatCanManage( plugin ) {
+		return ! plugin.wpcom &&
+			! plugin.sites.some( site => site.modules && site.modules.indexOf( 'manage' ) !== -1 );
+	},
+
+	getSelected() {
+		if ( ! this.props.plugins ) {
+			return [];
+		}
+		return this.props.plugins.filter( this.filterSelection.selected.bind( this ) );
+	},
+
+	siteSuffix() {
+		return ( this.props.selectedSite || this.props.sites.get().length === 1 ) ? '/' + this.props.sites.selected : '';
+	},
+
+	recordEvent( eventAction, includeSelectedPlugins ) {
+		eventAction += ( this.props.selectedSite ? '' : ' on Multisite' );
+		if ( includeSelectedPlugins ) {
+			const pluginSlugs = this.getSelected().map( plugin => plugin.slug );
+			analytics.ga.recordEvent( 'Plugins', eventAction, 'Plugins', pluginSlugs );
+		} else {
+			analytics.ga.recordEvent( 'Plugins', eventAction );
+		}
+	},
+
+	// Actions
+	toggleBulkManagement() {
+		const bulkManagement = ! this.state.bulkManagement;
+
+		this.setState( {
+			bulkManagement
+		} );
+
+		if ( bulkManagement ) {
+			this.recordEvent( 'Clicked Manage' );
+		} else {
```

Comment:
By returning this function, we can abort early and skip the whole `else` expression

```js
if ( bulkManagement ) {
	return this.recordEvent( 'Clicked Manage' );
}

this.setState( … );

…
```